mr_bots-engagement-service-twitter-clone.py

```diff
@@ -226,7 +226,17 @@ def generateTweets(consumer_key, consumer_secret, target, access_key, access_sec
         tweet = ' '.join(sent_tokes[0:-1]).replace(',','.')
     else:
         tweet = tweet
-    return tweet
+        
+        
+    already_tweeted = False
+    for t in tweet_texts:
+        if t.strip() == tweet.strip():
+            already_tweeted = True
+
+    if already_tweeted:
+        generateTweets(consumer_key, consumer_secret, target, access_key, access_secret)
+    else:
+        return tweet
 
 
```

Comment:
What I think we should do here is actually something like this, because idk it just feels right and frozenset is gonna be faster:

```
def check_tweet(tweet,tweet_texts):
    tweet_texts = frozenset([tweet_texts])
    existing_tweet = False
    clean_tweet = re.sub(r'\w+:\/{2}[\d\w-]+(\.[\d\w-]+)*(?:(?:\/[^\s/]*))*|\@\w+', '',tweet)
    if clean_tweet in tweet_texts:
        already_tweeted = True
    else:
        already_tweeted = False
    return already_tweeted

def generateTweets(consumer_key, consumer_secret, target, access_key, access_secret):
    markovLength = 2
    build_corpus(consumer_key, consumer_secret, target, access_key, access_secret)
    words = " ".join(str(x) for x in tweet_texts)
    buildMapping(wordlist(words), markovLength)
    tweet = genSentence(markovLength)
    sent_tokes = tweet.split()
    last_word = sent_tokes[-1]
    last_word = last_word.replace('.','').replace(',','')
    if last_word in articles:
        tweet = ' '.join(sent_tokes[0:-1])
    elif len(last_word) == 1:
        tweet = ' '.join(sent_tokes[0:-1]).replace(',','.')
    else:
        tweet = tweet

    if check_tweet(tweet) is True:
        generateTweets(consumer_key, consumer_secret, target, access_key, access_secret)
    else:
        return tweet
```

@ptsteadman is this RECURSIVE  ENOUGH FOR YOU ????????????????? JW. ASKING FOR A FRIEND.